src/kaleidoscope_internal/type_traits/has_method.h

```diff
@@ -23,46 +23,46 @@
+      /* if possible. If the method signature cannot be found,        */ __NL__ \
+      /* the substitution fails and the first version of method "test"*/ __NL__ \
+      /* will not be defined.                                         */ __NL__ \
+                                                                        __NL__ \
       template <typename ClassAux__>                                    __NL__ \
       static constexpr                                                  __NL__ \
-      /* The following decltype-clause defines the function return type __NL__ \
-       */                                                               __NL__ \
+      /* The following decltype-clause defines the function return type */ __NL__ \
+                                                                        __NL__ \
       decltype(                                                         __NL__ \
-         /* If &ClassAux__::METHOD_NAME exists and is of type           __NL__ \
-          * MethodType__, the list below evaluates to bool{} whose      __NL__ \
-          * type can be determined. Otherwise the comma expression      __NL__ \
-          * cannot be evaluated and the content                         __NL__ \
-          * of decltype is undefined and this function overload         __NL__ \
-          * is ignored by the compiler                                  __NL__ \
-          * (SFINAE = substitution failure is not an error)             __NL__ \
-          * and the test(...) overload is used instead.                 __NL__ \
-          */                                                            __NL__ \
+         /* If &ClassAux__::METHOD_NAME exists and is of type        */ __NL__ \
+         /* MethodType__, the list below evaluates to bool{} whose   */ __NL__ \
+         /* type can be determined. Otherwise the comma expression   */ __NL__ \
+         /* cannot be evaluated and the content                      */ __NL__ \
+         /* of decltype is undefined and this function overload      */ __NL__ \
+         /* is ignored by the compiler                               */ __NL__ \
+         /* (SFINAE = substitution failure is not an error)          */ __NL__ \
+         /* and the test(...) overload is used instead.              */ __NL__ \
+                                                                        __NL__ \
          static_cast<MethodType__>(                                     __NL__ \
             &ClassAux__::TEMPLATE_KEYWORD(UNWRAP TMPL_PARAM_LIST)       __NL__ \
                METHOD_NAME ADD_TEMPLATE_BRACES(UNWRAP TMPL_PARAM_LIST)  __NL__ \
          ), bool{}                                                      __NL__ \
       )                                                                 __NL__ \
-      test(int /* unused */)                                            __NL__ \
+      test(int /* unused */) /* NOLINT(readability/casting) */          __NL__ \
```

Comment:
cpplint would complain about C-style cating here, urging us to use C++-style `static_cast<>` instead. Considering this is a function argument, and unused at that, there's no point in any casting. Ignore the warning instead.